applications/luci-app-firewall/luasrc/model/cbi/firewall/zones.lua

```diff
@@ -32,6 +32,18 @@ for i, v in ipairs(p) do
 	v:value("ACCEPT", translate("accept"))
 end
 
+s:option(DummyValue, "offload_advice",
+	translate("Routing/NAT Offloading"),
+	translate("Exprimental feature. Requires kernel 4.14 and is not fully compatible with QoS/SQM."))
```

Comment:
Shouldn´t we check if flow offloding is available eg. with checking if "/sys/module/xt_FLOWOFFLOAD" exists instead of the hint with kernel 4.14 requirement?
As i know qos and sqm is only not supported with hw flow offloading -> move this to hw flow offloading option.

Reply:
> As i know qos and sqm is only not supported with hw flow offloading

Not quite sure of that...
See https://forum.lede-project.org/t/build-for-netgear-r7800/316/790

> > Enabling SQM and (software-) flowoffloading won’t break anything (hardware offloading might be more intrusive, but so far that’s only available for mt762x, not ipq806x or ar71xx), but with SQM enabled, there won’t be any flows that could be offloaded (as the kernel needs fine grained control over each frame). So yes, you can enable both, but flowoffloading loses and essentially disables itself if SQM is active; the net effect might be slightly worse, as the flowoffloading algorithm will continue to check each package if it can offload it, only to realize that it can’t.